astropy/coordinates/tests/test_separation.py

```diff
@@ -351,3 +354,66 @@ def test_return_types(coord_class, method, output_type):
     """
     coord = coord_class(0 * u.deg, 0 * u.deg, 1 * u.pc)
     assert type(getattr(coord, method)(coord)) is output_type
+
+
+@pytest.mark.parametrize("coord_class", [SkyCoord, ICRS])
+@pytest.mark.parametrize(
+    "frame_mismatch_kwarg,expectation",
+    [
+        pytest.param({"frame_mismatch": "ignore"}, nullcontext(), id="ignore"),
+        pytest.param(
+            {"frame_mismatch": "warn"},
+            pytest.warns(
+                AstropyUserWarning,
+                match=(
```

Comment:
This is a bit of a matter of preference, but I never think it is particularly to test the full error message. Here, it also makes the parametrization exceedingly long. Maybe just `match="transforming.*GCRS.*ICRS"`?

Reply:
The error messages are non-trivial, so it is good to test them, but it is indeed better to do that in a separate test, not here.